src/site/content/en/blog/serial/index.md

```diff
@@ -296,26 +296,45 @@ case, calling `reader.cancel()` will force `reader.read()` to resolve
 immediately with `{ value: undefined, done: true }` and therefore allowing the
 loop to call `reader.releaseLock()`.
 
-```js/15-18
+```js
 // Without transform streams.
 
-const reader = port.readable.getReader();
-
-// Listen to data coming from the serial device.
-while (true) {
-  const { value, done } = await reader.read();
-  if (done) {
-    reader.releaseLock();
-    break;
+let keepReading = true;
+let reader;
+
+async function readUntilClosed() {
+  while (port.readable && keepReading) {
+    reader = port.readable.getReader();
+    try {
+      while (true) {
+        const { value, done } = await reader.read();
+        if (done) {
+          // reader.cancel() has been called.
+          break;
+        }
+        // value is a Uint8Array.
+        console.log(value);
+      }
+    } catch (error) {
+      // Handle error...
+    } finally {
+      // Allow the serial port to be closed later.
+      reader.releaseLock();
+    }
   }
-  // value is a Uint8Array.
-  console.log(value);
+
+  await port.close();
 }
 
+const closed = readUntilClosed();
+
+
+// Sometimes later...
```

Comment:
"Sometime later"

It might be helpful to break this example into two code blocks with text in between making it even clearer that the code in this last section is expected to be invoked by something like a button click.

Reply:
@beaufortfrancois For what it's worth, I've made this exact suggestion on lots of code samples.

Reply:
Is https://github.com/GoogleChrome/web.dev/pull/4776/commits/887fdfdb3bda7a8460b2e30026e3541aebd92e05 better?